src/pkg/services/m365/api/channels_pager.go

```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
+		builder: c.Stable.
+			Client().
+			Teams().
+			ByTeamId(teamID).
+			Channels().
+			ByChannelId(channelID).
+			Messages().
+			Delta(),
+	}
+
+	return res
+}
+
+func (p *MessagePageCtrl) SetNext(nextLink string) {
+	p.builder = teams.NewItemChannelsItemMessagesDeltaRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *MessagePageCtrl) GetPage(ctx context.Context) (DeltaPageLinker, error) {
+	var (
+		resp DeltaPageLinker
+		err  error
+	)
+
+	resp, err = p.builder.Get(ctx, p.options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return resp, nil
+}
 
-// type messagePageCtrl struct {
-// 	gs      graph.Servicer
-// 	builder *teams.ItemChannelsItemMessagesRequestBuilder
-// 	options *teams.ItemChannelsItemMessagesRequestBuilderGetRequestConfiguration
-// }
+func (p *MessagePageCtrl) ValuesIn(l PageLinker) ([]models.ChatMessageable, error) {
+	return getValues[models.ChatMessageable](l)
+}
 
 // ---------------------------------------------------------------------------
 // channel pager
 // ---------------------------------------------------------------------------
 
 type ChannelDeltaEnumerator interface {
-	DeltaGetPager
+	PageLinker
 	ValuesInPageLinker[models.Channelable]
 	SetNextLinker
 }
 
 // TODO: implement
-// var _ ChannelDeltaEnumerator = &channelsPageCtrl{}
+var _ ChannelDeltaEnumerator = &channelPageCtrl{}
+
+type channelPageCtrl struct {
+	gs      graph.Servicer
+	builder *teams.ItemChannelsRequestBuilder
+	options *teams.ItemChannelsRequestBuilderGetRequestConfiguration
+}
+
+func (c Channels) NewChannelPager(
+	teamID,
+	channelID string,
+	fields []string,
+) *channelPageCtrl {
+	requestConfig := &teams.ItemChannelsRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsRequestBuilderGetQueryParameters{
+			Select: fields,
+		},
+	}
+
+	res := &channelPageCtrl{
+		gs:      c.Stable,
+		options: requestConfig,
+		builder: c.Stable.
+			Client().
+			Teams().
+			ByTeamId(teamID).
+			Channels(),
+	}
 
-// type channelsPageCtrl struct {
-// 	gs      graph.Servicer
-// 	builder *teams.ItemChannelsChannelItemRequestBuilder
-// 	options *teams.ItemChannelsChannelItemRequestBuilderGetRequestConfiguration
-// }
+	return res
+}
+
+func (p *channelPageCtrl) SetNext(nextLink string) {
+	p.builder = teams.NewItemChannelsRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *channelPageCtrl) GetPage(ctx context.Context) (PageLinker, error) {
+	var (
+		resp PageLinker
+		err  error
+	)
+
+	resp, err = p.builder.Get(ctx, p.options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return resp, nil
+}
+
+func (p *channelPageCtrl) ValuesIn(l PageLinker) ([]models.Channelable, error) {
+	return getValues[models.Channelable](l)
+}
+
+func (p *channelPageCtrl) GetOdataNextLink() *string {
+	// No next link preent in the API result
+	emptyString := ""
+	return &emptyString
+}
+
+// ---------------------------------------------------------------------------
+// non delta item pager
```

Comment:
This should get moved up to co-locate with the other message pager.  The current page layout goes `messages > channels > messages`, which doesn't flow very well.
```suggestion
// non delta channel message pager
```